spec/unit/face/config_spec.rb

```diff
@@ -41,6 +46,29 @@ def render(action, result)
     expect(render(:print, result)).to eq("file\n")
   end
 
+  it "prints the section and environment, and not a warning,  when a section is given" do
+    Puppet.settings.parse_config(<<-CONF)
+    [user]
+    syslogfacility = file
+    CONF
+
+    subject.expects(:warn_default_section).never
+    subject.expects(:report_section_and_environment).once
+
+    result = subject.print("syslogfacility", :section => "user")
+    expect(render(:print, result)).to eq("file\n")
+  end
+
+  it "prints a warning and the section and environment when no section is given" do
+    Puppet[:trace] = true
+
+    subject.expects(:warn_default_section).once
+    subject.expects(:report_section_and_environment).once
+
+    result = subject.print("trace")
+    expect(render(:print, result)).to eq("true\n")
```

Comment:
We should add similar tests for `set` and `delete`, since you've added this functionality to those actions as well